core/src/main/java/bio/terra/pearl/core/service/portal/MailingListContactService.java

```diff
@@ -22,6 +27,17 @@ public Optional<MailingListContact> findByPortalEnv(UUID portalEnvId, String ema
         return dao.findByPortalEnv(portalEnvId, emailAddress);
     }
 
+    @Transactional
+    public List<MailingListContact> bulkCreate(UUID portalEnvId, List<MailingListContact> contacts, DataAuditInfo auditInfo) {
+        // remove contacts from the submitted list if they already exist in this environment
+        findByPortalEnv(portalEnvId).forEach(existing ->
+                contacts.removeIf(contact -> contact.getEmail().equals(existing.getEmail())));
+
+        contacts.forEach(contact -> contact.setPortalEnvironmentId(portalEnvId));
+        bulkCreate(contacts, auditInfo);
+        return contacts;
+    }
+
     @Transactional
     public void deleteByPortalEnvId(UUID portalEnvId) {
```

Comment:
since it's data audited, should this delete take in a DataAuditInfo and be audited?

Reply:
I could go either way on this. Since we'd only ever use this if we're destroying the portal, I'm not sure it's worth keeping a record of each contact that was also destroyed, especially since this is sort of internal-only during portal repopulate. But if you or Devon feel strongly, I can totally make the change